modules/biobank/php/container.class.inc

```diff
@@ -0,0 +1,339 @@
+ * permanently store and retrieve Container Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category Loris
+ * @package  Biobank
+ * @author   Henri Rabalais <hrabalais.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+class Container implements \JsonSerializable, \LORIS\Data\DataInstance
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a container entity.
+     *
+     * @access private
+     */
+    private $id;                 //int
+    private $barcode;            //int
+    private $typeId;             //int
+    private $dimensionId;        //int
+    private $temperature;        //float
+    private $statusId;           //int
+    private $originId;           //int
+    private $centerId;           //int
+    private $parentContainerId;  //int
+    private $childContainerIds;  //array
+    private $coordinate;         //int
+
+    /**
+     * Initiliazes a new instance of the Container Class
+     */
+    function __construct()
+    {
+    }
+
+    /**
+     * Sets the container's ID
+     *
+     * @param int $id the container's ID
+     *
+     * @return void
+     */
+    public function setId(?int $id = null) : void
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * Gets the container's ID
+     *
+     * @return int
+     */
+    public function getId() : ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * Sets the container's barcode
+     *
+     * @param string $barcode the container's barcode
+     *
+     * @return void
+     */
+    public function setBarcode(string $barcode) : void
+    {
+        $this->barcode = $barcode;
+    }
+
+    /**
+     * Gets the container's barcode
+     *
+     * @return string
+     */
+    public function getBarcode() : string
+    {
+        return $this->barcode;
+    }
+
+    /**
+     * Sets the ID of the container type
+     *
+     * @param int $typeId ID of the container type
+     *
+     * @return void
+     */
+    public function setTypeId(int $typeId) : void
+    {
+        $this->typeId = $typeId;
+    }
+
+    /**
+     * Gets the ID of the container type
+     *
+     * @return int
+     */
+    public function getTypeId() : int
+    {
+        return $this->typeId;
+    }
+
+    /**
+     * Sets the ID of the container's dimensions
+     *
+     * @param int $dimensionId ID of the container's dimensions
+     *
+     * @return void
+     */
+    public function setDimensionId(int $dimensionId) : void
+    {
+        $this->dimensionId = $dimensionId;
+    }
+
+    /**
+     * Gets the ID of the container's dimensions
+     *
+     * @return int
+     */
+    public function getDimensionId() : int
+    {
+        return $this->dimensionId;
+    }
+
+    /**
+     * Sets the container's temperature
```

Comment:
Celsius?

Reply:
I'm not entirely sure for the moment. It may be wise to allow the Project to configure the temperature unit. 
@ridz1208 ?

Reply:
good question. def a project decision

Reply:
I don't think it would be wise to allow the project to configure units that are used internally.

Reply:
No? So always use celcius?

Reply:
I think it would make sense to use Celsius internally and do a front-end conversion to display Fahrenheit if this is a requested feature for users.

Reply:
That makes sense. In that case should we just go ahead and use Kelvin in the back end? Maybe we can decide that at a later point @ridz1208. I'll set everything to Celsius for now.